File: solutions/go/luhn/1/luhn.go
package luhn

import (
    "unicode"
)

func Valid(id string) bool {
    digits := []rune{}
    for _, r := range id {
        if unicode.IsSpace(r) {
            continue
        }
        if !unicode.IsDigit(r) {
            return false
        }
        digits = append(digits, r)
    }

    if len(digits) <= 1 {
        return false
    }

    sum := 0
    double := false

    // Process digits from right to left
    for i := len(digits) - 1; i >= 0; i-- {
        digit := int(digits[i] - '0')
        if double {
            digit *= 2
            if digit > 9 {
                digit -= 9
            }
        }
        sum += digit
        double = !double
    }

    return sum%10 == 0
}

File: solutions/csharp/luhn/1/Luhn.cs
public static class Luhn
{
    public static bool IsValid(string number)
    {
        var digits = new List<int>();
        foreach (char ch in number)
        {
            if (Char.IsWhiteSpace(ch))
                continue;
            if (!Char.IsDigit(ch))
                return false;
            digits.Add(ch - '0');
        }

        if (digits.Count <= 1)
            return false;

        int sum = 0;
        bool doubleFlag = false;

        for (int i = digits.Count -1; i >= 0; i--)
        {
            int digit = digits[i];
            if (doubleFlag)
            {
                digit *= 2;
                if (digit > 9)
                    digit -= 9;
            }
            sum += digit;
            doubleFlag = !doubleFlag;
        }
        return sum % 10 == 0;
    }
}